### PR TITLE
GH-13321: Move Order Normalizers to lib

### LINKED ISSUE #13321: Order Normalizers move out to Order Lib
* Move `ORDER_NORMALIZER` and `REPLENISHMENT_ORDER_NORMALIZER`, their implementation and their provider config in the order-lib.
They are in `projects/core/src/occ/adapters/order`

* Move `OrderEntryPromotionsService` in cart lib.  Consider different options if this  creates a dependency on the whole cart lib: duplicate the code, or perhaps create standalone functions instead.

* Evaluate moving `ORDER_NORMALIZER` and `REPLENISHMENT_ORDER_NORMALIZER` tokens out of checkout lib and into order lib with their default implementation

Hint: Refer to [this PR](https://github.com/SAP/spartacus/pull/13361/files) for an overview of the impacted code. This is the PR where we reverted the move of the Order Normalizers in Order Lib.

### PR DESCRIPTION
**Pull request by plabadie**

### CODE REVIEW

File: projects/core/src/occ/adapters/order/order-occ.module.ts [hunk 1]
```diff
@@ -1,23 +1,8 @@
...
-      provide: ORDER_NORMALIZER,
-      useExisting: OccOrderNormalizer,
-      multi: true,
-    },
-    {
-      provide: REPLENISHMENT_ORDER_NORMALIZER,
-      useExisting: OccReplenishmentOrderNormalizer,
-      multi: true,
-    },
-  ],
```
Comment (WeizhengSap) [inline projects/core/src/occ/adapters/order/order-occ.module.ts:21]: can we just remove this module?

File: projects/core/src/occ/adapters/user/index.ts [hunk 2]
```diff
@@ -5,9 +5,5 @@ export * from './occ-user-address.adapter';
 export * from './occ-user-consent.adapter';
 export * from './occ-user-interests.adapter';
 export * from './occ-user-notification-preference.adapter';
-export * from './occ-user-order.adapter';
 export * from './occ-user-payment.adapter';
-export * from './occ-user-replenishment-order.adapter';
-export * from './user-occ-transitional.module';
 export * from './user-occ-transitional_4_2.module';
```
Comment (WeizhengSap) [inline projects/core/src/occ/adapters/user/index.ts:9]: In `UserOccModule`, there are some order related providers, can we remove them?

File: feature-libs/order/occ/adapters/converters/occ-order-normalizer.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import { Injectable } from '@angular/core';
-import { PromotionResult } from '../../../../model/cart.model';
-import { Order, OrderEntry } from '../../../../model/order.model';
-import { PRODUCT_NORMALIZER } from '../../../../product/connectors/product/converters';
+import { OrderEntryPromotionsService } from '@spartacus/cart/main/occ';
```
Comment (WeizhengSap) [inline feature-libs/order/occ/adapters/converters/occ-order-normalizer.ts:2]: It's better not import `cart occ` in `order occ`. We can create a normalizer for promotions, convert this  `OrderEntryPromotionsService` to a normalizer, or create a token for this service.

File: projects/core/src/occ/adapters/order/order-occ.module.ts (hunk 1 above)
Comment (plabadie) [reply projects/core/src/occ/adapters/order/order-occ.module.ts:21]: Yes.  It its removed now.

File: feature-libs/order/occ/adapters/converters/occ-order-normalizer.ts (hunk 3 above)
Comment (plabadie) [reply feature-libs/order/occ/adapters/converters/occ-order-normalizer.ts:2]: I created follow up ticket #14093 to address this

**Review by WeizhengSap — APPROVED** [review]

**Pull request merged**